lib/src/protocol/validator.dart

```diff
@@ -0,0 +1,74 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:json_schema/json_schema.dart';
+import 'package:path/path.dart' as p;
+
+class Validator {
+  factory Validator._internal() {
+    _initialize();
+    return Validator._();
+  }
+
+  static final Map<String, JsonSchema> _schemaMap = {};
+  static final Validator _instance = Validator._internal();
+
+  static void validate(Map<String, dynamic> json, String schemaName) {
+    _instance.validateJson(json, schemaName);
+  }
+
+  Validator._();
```

Comment:
nit: but maybe we should move this up to just below the `_instance` definition to keep related things together?